cmd/mapp-worker/main.go

```diff
@@ -0,0 +1,392 @@
+package main
+
+import (
+	"bytes"
+	"crypto/sha512"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/hex"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"os/signal"
+	"regexp"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/robfig/cron"
+	log "github.com/sirupsen/logrus"
+)
+
+var letsencryptPem = `-----BEGIN CERTIFICATE-----
```

Comment:
I agree.

In the general case, my answer would be "users should just install LetsEncrypt's root certificate". This is not possible on DICE. LE websites work on DICE because browsers have their own cert location.

So we need to hold a copy of the file somewhere for convenience. Putting it in a separate file _does_ work, but it decreases the portability of mapp-worker (now you need to always have reference to a root cert).

Currently no config file is required, just set some environment variables and run the worker. Needing to carry a cert folder on DICE is a bit of extra maintenance (although one-time).

There are problems with this approach, though:
- The certificate in there is a combination of two [certificates](https://letsencrypt.org/certificates/):
    - [ISRG Root X1 (self-signed)](https://letsencrypt.org/certs/isrgrootx1.pem.txt)
    - some other certificate which according to Google has expired (by searching for a sequence of characters inside the cert)
- ^^^^^^ it should not be trusting an expired certificate (but may it's already smart enough to reject it based on expiry date?)
- it is not protected against revoked certs (unless it's already smart enough to handle this)
- simply not future proof

To be honest, I'm hoping that by now DICE will trust LE. So, next steps here would be:

- [ ] Check to see if LE works on DICE without this thing
- [ ] If it does not work, contact Computing Support and ask them to install the root certificate
- [ ] Otherwise, on mapp-worker load, download root certificate from LetsEncrypt's website

The proposed alternate solution (downloading from LE's website) is good because:
- always up to date
- handles revoked certs properly

It may not work if LE is encrypted using LE certs though :^) <sub>hint: it is</sub>